q2_diversity/_beta/_method.py

```diff
@@ -108,6 +75,24 @@ def beta_phylogenetic_alt(table: BIOMV210Format, phylogeny: NewickFormat,
     else:
         f = metrics[metric]
 
+    # avoid a full parse of the table just to check shape and IDs
```

Comment:
This... sucks. To preserve the existing test code, we needed to a) test for an empty table and b) assert the table is represented by the tree. Both are useful. Both need some additional effort here. The logic below can be refactored, and perhaps it just makes sense to do a full parse of the BIOM table in order to avoid h5py directly. Happy to go any advised direction here.